### ISSUE TITLE
Bug: Project Fetch Fails Initializing Rally

### ISSUE THREAD

**Issue by ClockerZadq** [RallyTools/RallyRestToolkitForPython#88]
Hi,

I'm using the latest version of this tool and I have not been able to successfully connect to the Rally instance and utilize the API. My script is failing with 

`RallyRESTAPIERROR: the current Workspace |My Workspace| does not contain a Project that matches the current setting of the Project: None.`

I init the Rally class by using a rally-v2.0.cfg with the appropriate options set. I have tested it and the file gets parsed correctly.

From what I've discovered looking at source, the lib is attempting to identify if None exists in the workspace, throwing the error above if it doesn't. The None tells me that the project var has been assigned as a Null object. Why doesn't it use the project var that was specified in the config file?

**Comment by ClockerZadq** [comment]
I added the following line to context.py

296 + self._currentProject = project_name
297    result = self.agent.get('Project', fetch='Name', workspace=self._currentWorkspace)

I'm not very familiar with this API but the script now runs and it looks like I'm able to use it. I guess... cheers!

**Comment by klehman-rally** [comment]
If you run the unmodified pyral 1.2.0 code using the rallyfire.py script and your rally-v2.0.cfg file, does it result in the same RallyRESTAPIError ?  If so, would you be able to "fuzz" the values in your rally-v2.0.cfg (so that you are not supplying the actual credentials, although it would be helpful to preserve the WORKSPACE  and PROJECT values to see if there's any issue with punctuation chars, spaces or multibyte characters...) and post that here?  Are you using the rallyWorkset function prior to calling the Rally constructor with your parameters?  I attempted to repro what you saw using various configurations of PROJECT in my rally-v2.0.cfg and did not see the result you observed.  It would help me implement a general fix for this if I could repro your observations.

**Comment by mattguymad** [comment]
Hi,

I have the similar error. I just instantiate the Rally class via:

Rally(server, apikey = apiKey, workspace=workspace, project=parentProject)

And I encountered that problem. Zaq's fix solved the problem for me.
Cheers,
Matt

**Comment by ClockerZadq** [comment]
@klehman-rally I did not try the rallyfire.py script, what would that show? I do call the rallyWorkset function before initiating the Rally object, though I also tried without using it which gave the same results. It seems to be that there is nothing wrong with how the configuration is being read or the values presented to it. 

The stack trace that gets presented to me was the same as issue 91 leading me to believe that the function is attempting to verify that the project (the set default project) is part of the the projects belonging to the worspace. However, the project that is being evaluated against the list of projects for that workspace is None because a default project has not been assigned. That is what my solution does, sets the project so that when it is accessed it doesnt result to None. I don't think there's much more to this ?

Glad I could help @mattguymad!
